overlord/devicestate/devicestate_cloudinit_test.go

```diff
@@ -56,6 +59,169 @@ func (s *cloudInitBaseSuite) SetUpTest(c *C) {
 	c.Assert(err, IsNil)
 }
 
+type cloudInitUC20Suite struct {
+	cloudInitBaseSuite
+}
+
+var _ = Suite(&cloudInitUC20Suite{})
+
+func (s *cloudInitUC20Suite) SetUpTest(c *C) {
+	s.cloudInitBaseSuite.SetUpTest(c)
+
+	// make a uc20 style dangerous model assertion for the device
+	// note that actually the devicemgr ensure only cares about having a grade
+	// for uc20, it doesn't use the grade for anything right now, the install
+	// handler code however does care about the grade, so here we just default
+	// to signed
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	s.makeModelAssertionInState(c, "canonical", "pc20-model", map[string]interface{}{
+		"display-name": "UC20 pc model",
+		"architecture": "amd64",
+		"base":         "core20",
+		"grade":        "signed",
+		"snaps": []interface{}{
+			map[string]interface{}{
+				"name":            "pc-kernel",
+				"id":              "pckernelidididididididididididid",
+				"type":            "kernel",
+				"default-channel": "20",
+			},
+			map[string]interface{}{
+				"name":            "pc",
+				"id":              "pcididididididididididididididid",
+				"type":            "gadget",
+				"default-channel": "20",
+			}},
+	})
+	devicestatetest.SetDevice(s.state, &auth.DeviceState{
+		Brand:  "canonical",
+		Model:  "pc20-model",
+		Serial: "serial",
+	})
+}
+
+func (s *cloudInitSuite) SetUpTest(c *C) {
```

Comment:
this feels oddly placed now

Reply:
do you mean just where this function is physically located in the file? or the code in this function is oddly place and should be in some other setup function?

Reply:
yes, I mean the placement of the function itself